src/Concordium/Client/GRPC2.hs

```diff
@@ -0,0 +1,2766 @@
+        irMin <- fromProto =<< iRange ^? ProtoFields.min
+        irMax <- fromProto =<< iRange ^? ProtoFields.max
+        return Parameters.InclusiveRange{..}
+
+instance FromProto Proto.CapitalBound where
+    type Output' Proto.CapitalBound = Parameters.CapitalBound
+    fromProto cBound = fmap Parameters.CapitalBound . fromProto =<< cBound ^? ProtoFields.value
+
+instance FromProto Proto.LeverageFactor where
+    type Output' Proto.LeverageFactor = Parameters.LeverageFactor
+    fromProto lFactor = fmap Parameters.LeverageFactor . fromProto =<< lFactor ^? ProtoFields.value
+
+instance FromProto Proto.PoolParametersCpv1 where
+    type Output' Proto.PoolParametersCpv1 = (Parameters.PoolParameters 'ChainParametersV1)
+    fromProto pParams = do
+        _ppPassiveCommissions <- do
+            _finalizationCommission <- fromProto =<< pParams ^? ProtoFields.passiveFinalizationCommission
+            _bakingCommission <- fromProto =<< pParams ^? ProtoFields.passiveBakingCommission
+            _transactionCommission <- fromProto =<< pParams ^? ProtoFields.passiveTransactionCommission
+            return CommissionRates{..}
+        _ppCommissionBounds <- do
+            cBounds <- pParams ^? ProtoFields.commissionBounds
+            _finalizationCommissionRange <- fromProto =<< cBounds ^? ProtoFields.finalization
+            _bakingCommissionRange <- fromProto =<< cBounds ^? ProtoFields.baking
+            _transactionCommissionRange <- fromProto =<< cBounds ^? ProtoFields.transaction
+            return Parameters.CommissionRanges{..}
+        _ppMinimumEquityCapital <- fromProto =<< pParams ^? ProtoFields.minimumEquityCapital
+        _ppCapitalBound <- fromProto =<< pParams ^? ProtoFields.capitalBound
+        _ppLeverageBound <- fromProto =<< pParams ^? ProtoFields.leverageBound
+        return Parameters.PoolParametersV1{..}
+
+instance FromProto Proto.Sha256Hash where
+    type Output' Proto.Sha256Hash = Hash
+    fromProto = deMkSerialize
+
+instance FromProto Proto.ProtocolUpdate where
+    type Output' Proto.ProtocolUpdate = Updates.ProtocolUpdate
+    fromProto pUpdate = do
+        puMessage <- pUpdate ^? ProtoFields.message
+        puSpecificationURL <- pUpdate ^? ProtoFields.specificationUrl
+        puSpecificationHash <- fromProto =<< pUpdate ^? ProtoFields.specificationHash
+        puSpecificationAuxiliaryData <- pUpdate ^? ProtoFields.specificationAuxiliaryData
+        return Updates.ProtocolUpdate{..}
+
+instance FromProto Proto.TransactionFeeDistribution where
+    type Output' Proto.TransactionFeeDistribution = Parameters.TransactionFeeDistribution
+    fromProto tfDistribution = do
+        _tfdBaker <- fromProto =<< tfDistribution ^? ProtoFields.baker
+        _tfdGASAccount <- fromProto =<< tfDistribution ^? ProtoFields.gasAccount
+        return Parameters.TransactionFeeDistribution{..}
+
+instance FromProto Proto.UpdatePayload where
+    type Output' Proto.UpdatePayload = (Updates.UpdateType, Updates.UpdatePayload)
```

Comment:
There is a function
```
updateType :: UpdatePayload -> UpdateType
```
in base.

Please use that and do not reinvent this here since that is error prone. Just return UpdatePayload.